crates/core/src/operations/unset_tbl_properties.rs

```diff
@@ -0,0 +1,169 @@
+//! Unset table properties of the table
+
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use chrono::Utc;
+use futures::future::BoxFuture;
+use serde_json::json;
+
+use crate::kernel::{Action, CommitInfo, IsolationLevel, Protocol};
+use crate::logstore::LogStoreRef;
+use crate::operations::transaction::commit;
+use crate::protocol::DeltaOperation;
+use crate::table::config::DeltaConfigKey;
+use crate::table::state::DeltaTableState;
+use crate::DeltaTable;
+use crate::{DeltaResult, DeltaTableError};
+
+/// Remove constraints from the table
+pub struct UnsetTablePropertiesBuilder {
+    /// A snapshot of the table's state
+    snapshot: DeltaTableState,
+    /// Name of the property
+    properties: Option<Vec<String>>,
+    /// Raise if property doesn't exist
+    raise_if_not_exists: bool,
+    /// Delta object store for handling data files
+    log_store: LogStoreRef,
+    /// Additional metadata to be added to commit
+    app_metadata: Option<HashMap<String, serde_json::Value>>,
+}
+
+impl UnsetTablePropertiesBuilder {
+    /// Create a new builder
+    pub fn new(log_store: LogStoreRef, snapshot: DeltaTableState) -> Self {
+        Self {
+            properties: None,
+            raise_if_not_exists: true,
+            snapshot,
+            log_store,
+            app_metadata: None,
+        }
+    }
+
+    /// Specify the constraint to be removed
+    pub fn with_property(mut self, name: String) -> Self {
+        match self.properties {
+            Some(mut vec_name) => {
+                vec_name.push(name);
+                self.properties = Some(vec_name);
+            }
+            None => self.properties = Some(vec![name]),
+        }
+        self
+    }
```

Comment:
here is was a little bit unsure what behaviour I would expect. From the name, my first instinct was that this would set the property, not append it to any existing ones.

What do you think?

Reply:
So, I was thinking people could either chain multiple `.with_property().with_property()` to add the configkeys or they use the `with_properties `to pass a Vec.